qc/homo_sapiens_qc.py

```diff
@@ -366,4 +366,149 @@ def __init__(self):
             # NEAN pop. coalesces into AF (E7)
             msprime.MassMigration(
                 time=T_NEAN_split, source = 3, dest = 0, proportion=1.0)
-        ]
\ No newline at end of file
+        ]
+
+
+class KammAncientSamples(models.Model):
+    """
+    Demographic inferred by momi described in Kamm et al. (2019). The model is
+    illustrated in Figure 3, with parameters given in Table 2.
+    """
+    def __init__(self):
+        super().__init__()
+
+        generation_time = 25
+
+        # population sizes
+        N_Losch = 1.92e3
+        N_Mbu = 1.73e4
+        N_Mbu_Losch = 2.91e4
+        N_Han = 6.3e3
+        N_Han_Losch = 2.34e3
+        N_Nean_Losch = 1.82e4
+        N_Nean = 86.9
+        N_LBK = 75.7
+        N_Sard = 1.5e4
+        N_Sard_LBK = 1.2e4
+
+        # unknown population sizes
+        # these should be set to ancestral Eurasian population size,
+        # but at the moment it's unclear to me if that is N_Han_Losch? N_Losch?
+        N_Basal = N_Losch
+        N_Ust = N_Basal
+        N_MA1 = N_Basal
+
+        # population merge times in years, divided by generation time
+        t_Mbu_Losch = 9.58e4 / generation_time
+        t_Han_Losch = 5.04e4 / generation_time
+        t_Ust_Losch = 5.15e4 / generation_time
+        t_Nean_Losch = 6.96e5 / generation_time
+        t_MA1_Losch = 4.49e4 / generation_time
+        t_LBK_Losch = 3.77e4 / generation_time
+        t_Basal_Losch = 7.98e4 / generation_time
+        t_Sard_LBK = 7.69e3 / generation_time
+        # t_GhostWHG_Losch = 1.56e3 / generation_time
+
+        # pulse admixture times and fractions
+        p_Nean_to_Eur = 0.0296
+        t_Nean_to_Eur = 5.68e4 / generation_time
+        p_Basal_to_EEF = 0.0936
+        t_Basal_to_EEF = 3.37e4 / generation_time
+        p_GhostWHG_to_Sard = 0.0317
+        t_GhostWHG_to_Sard = 1.23e3 / generation_time
+
+        # sample_times (in years), divided by estimated generation time
+        # t_Mbuti = 0
+        # t_Han = 0
+        # t_Sardinian = 0
+        # t_Loschbour = 7.5e3 / generation_time
+        # t_LBK = 8e3 / generation_time
+        # t_MA1 = 24e3 / generation_time
+        # t_UstIshim = 45e3 / generation_time
+        t_Altai = 50e3 / generation_time
```

Comment:
This is okay for now but it will be a problem when I add a check for consistency of sampling times during the model comparison tests. These sampling times will have to be passed to the population configurations.

Reply:
@ndukler I'll change these now.